### PR TITLE
Replaced asserts with exceptions

### PR DESCRIPTION
**Pull request by ArthurDent62**
According to issue #10

### CODE REVIEW

**Comment by caternuson** [comment]
Looks like Travis is failing on a few pylint things. Can you check them a push a new commit with fixes.

**Comment by ArthurDent62** [comment]
Travis was lazy last night (my time) and after 15 minutes or so I gave up. Let's see if he likes my changes now.

**Review by makermelissa — APPROVED** [review]
Code looks good and Travis is passing now, however I don't have the hardware to test this.

File: adafruit_tlc5947.py [hunk 1]
```diff
@@ -104,7 +104,8 @@ def duty_cycle(self):
 
         @duty_cycle.setter
         def duty_cycle(self, val):
-            assert 0 <= val <= 65535
+            if (val < 0) or (val > 65535):
```
Comment (caternuson) [inline adafruit_tlc5947.py]: Remove `()`s. The "pythonic" syntax is:
```python
if val < 0 or val > 65535:
```
Same for all other conditionals.

File: adafruit_tlc5947.py [hunk 2]
```diff
@@ -128,6 +129,8 @@ def frequency(self, val):
 
 
     def __init__(self, spi, latch, *, auto_write=True, num_drivers=1):
+        if num_drivers < 1:
+            raise ValueError("need at least one driver; {0} is not supported.".format(num_drivers))
```
Comment (caternuson) [inline adafruit_tlc5947.py]: Capitalize first word in message.
```
Need at least...
```
Same for all messages.

**Review by caternuson — CHANGES_REQUESTED** [review]

**Comment by caternuson** [comment]
Thanks for putting this back on the radar. I've got a few stylistic change requests.

**Comment by makermelissa** [comment]
No problem. Now that you pointed those out, I agree.

**Review by caternuson — APPROVED** [review]

**Pull request merged**